File: Assignment_2/task_2.cpp
// Implementing Swap Function in C++

#include<stdio.h>
#include<stdlib.h>


void swapByValue(int x, int y) {

    int swap = x;
    x = b;
    b = swap;

}


void swapByReference(int *x, int *y) {

    int swap = *x;
    *x = *y;
    *y = swap;
}

void swapByPointer(int &x, int &y) {

    int swap = x;
    x = y;
    y = swap;
}



int main() {

    int a = 3, b = 4;

    printf("Before Swap");

    printf("a = %d, b = %d \n", a, b);

    printf("After Swap by Value ");

    swapByValue(a,b);

    printf("a = %d, b = %d \n", a, b);

    printf("After Swap by Reference ");

    swapByReference(a,b);

    printf("a = %d, b = %d \n", a, b);

    printf("After Swap by Pointer ");

    swapByPointer(a,b);

    printf("a = %d, b = %d \n", a, b);
}

File: Assignment_2/task_2.java
import java.io.*;

public class task_2 {

    public static void swapByValue(int x, int y){

        int swap = x;
        x = y;
        y = swap;
    }   


    public static void swapByReference(IntWrapper x, IntWrapper y) {

    IntWrapper swap = new IntWrapper(x.x);
      x.x = y.x;
      y.x = swap.x;
    }

    public static void main(String[] args) {
    
    // Swap by Value

      int x = 35;
      int y = 25;
      System.out.println("Before swapping, x = " + x + " and y = " + y);
      
      swapByValue(x, y);
      
      System.out.println("After swapping, x = " + x + " and y is " + y);

    // Swap ByRefrence 
    
      IntWrapper x = new IntWrapper(30);
      IntWrapper y = new IntWrapper(45);
     
      System.out.println("Before swapping, x = " + x.x + " and y = " + y.x);
    
      swapByReference(x, y);
      
      System.out.println("After swapping, x = " + x.x + " and b is " + y.x);
   }
}

class IntWrapper {
   public int x;
   public IntWrapper(int x){ this.x = x;}
}